onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -168,7 +165,18 @@ void GetPyObjFromTensor(const Tensor& rtensor, py::object& obj) {
       PyArray_DATA(reinterpret_cast<PyArrayObject*>(obj.ptr())));
 
   if (numpy_type != NPY_OBJECT) {
-    memcpy(outPtr, rtensor.DataRaw(dtype), dtype->Size() * shape.Size());
+    //if it is not cpu tensor, need to copy to host
+    if (rtensor.Location().device.Type() != OrtDevice::CPU) {
+      if (!data_transfer_manager)
+        throw std::runtime_error("GetPyObjFromTensor: data transfer manager is needed when convert non-CPU tensor to numpy array");
+      static const OrtMemoryInfo cpu_alloc_info{onnxruntime::CPU, OrtDeviceAllocator};
```

Comment:
nit: 'to convert' or 'when converting'

What provides a data_transfer_manager instance to this function (i.e. if you see this error how can  you fix it)?

Reply:
Currently the caller that provides a data_transfer_manager is from TrainingSession. It is called to return a checkpoint state to the python script. Checkpoint state contains, for example, model weights which are in GPU memory. That is why a data_transfer_manager is needed. 

GetPyObjFromTensor is also called within InferenceSession. In these cases, it is to obtain fetch values which are already in CPU memory already. data_transfer_manager is not needed in these cases.

It is invalid to call GetPyObjFromTensor with non-CPU tensor without providing a data_transfer_manager. Therefore the error message.